src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
```

Comment:
The variable name `reorderedDate` is misleading since it's not necessarily reordered yet. A more suitable name could be `newDateParts`.